format/FlightSql.proto

```diff
@@ -933,6 +933,16 @@ message CommandGetDbSchemas {
  *  [optional] table_schema: bytes not null (schema of the table as described in Schema.fbs::Schema,
  *                                           it is serialized as an IPC message.)
  * >
+ * Fields on table_schema may contain the following metadata:
+ *  - CATALOG_NAME      - Table's catalog name
+ *  - SCHEMA_NAME       - Table's schema name
+ *  - TABLE_NAME        - Table name
+ *  - PRECISION         - Column precision/size
+ *  - SCALE             - Column scale/decimal digits
+ *  - IS_AUTO_INCREMENT - "1" if column is auto incremented, "0" otherwise.
+ *  - IS_CASE_SENSITIVE - "1" if column is case sensitive, "0" otherwise.
+ *  - IS_READ_ONLY      - "1" if column is read only, "0" otherwise.
+ *  - IS_SEARCHABLE     - "1" if column is searchable, "0" otherwise.
```

Comment:
Assuming that if a particular metadata is not known or not supported, it should just be omitted?

Reply:
Can we namespace these values? e.g. `arrow.flight.sql.catalog_name`.

Or actually, in analogy with [extension types](https://arrow.apache.org/docs/format/Columnar.html#extension-types) it should be something like `ARROW:FLIGHT:SQL:catalog_name`

Reply:
Can we also define what "searchable" means in this context?